### PR TITLE
docker: updated lnd/Dockerfile

### PR DESCRIPTION
**Pull request by tuxx42**
This commit fixes an issue with the lnd/Dockerfile using golang 1.10 instead of the recommended 1.11 which results in the following error

```
Step 5/12 : RUN apk add --no-cache     git     make &&  cd /go/src/github.com/lightningnetwork/lnd &&  make &&  make install
 ---> Running in 8d04c3b0e864
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
(1/7) Installing nghttp2-libs (1.32.0-r0)
(2/7) Installing libssh2 (1.8.0-r3)
(3/7) Installing libcurl (7.61.1-r1)
(4/7) Installing expat (2.2.5-r0)
(5/7) Installing pcre2 (10.31-r0)
(6/7) Installing git (2.18.1-r0)
(7/7) Installing make (4.2.1-r2)
Executing busybox-1.28.4-r1.trigger
OK: 19 MiB in 21 packages
\033[0;32m Building debug lnd and lncli.\033[0m
GO111MODULE=on go build -v -tags="dev" -o lnd-debug -ldflags "-X github.com/lightningnetwork/lnd/build.Commit=v0.5.1-beta-191-g0fafd5e2fd824f38ec6a03a56488de9c0798f34f" github.com/lightningnetwork/lnd
nat/upnp.go:9:2: cannot find package "github.com/NebulousLabs/go-upnp" in any of:
        /usr/local/go/src/github.com/NebulousLabs/go-upnp (from $GOROOT)
        /go/src/github.com/NebulousLabs/go-upnp (from $GOPATH)
aezeed/cipherseed.go:12:2: cannot find package "github.com/Yawning/aez" in any of:
        /usr/local/go/src/github.com/Yawning/aez (from $GOROOT)
        /go/src/github.com/Yawning/aez (from $GOPATH)
breacharbiter.go:11:2: cannot find package "github.com/btcsuite/btcd/blockchain" in any of:
        /usr/local/go/src/github.com/btcsuite/btcd/blockchain (from $GOROOT)
        /go/src/github.com/btcsuite/btcd/blockchain (from $GOPATH)
fundingmanager.go:11:2: cannot find package "github.com/btcsuite/btcd/btcec" in any of:
        /usr/local/go/src/github.com/btcsuite/btcd/btcec (from $GOROOT)
        /go/src/github.com/btcsuite/btcd/btcec (from $GOPATH)
chainntnfs/interface.go:7:2: cannot find package "github.com/btcsuite/btcd/btcjson" in any of:
        /usr/local/go/src/github.com/btcsuite/btcd/btcjson (from $GOROOT)
        /go/src/github.com/btcsuite/btcd/btcjson (from $GOPATH)
chainparams.go:4:2: cannot find package "github.com/btcsuite/btcd/chaincfg" in any of:
        /usr/local/go/src/github.com/btcsuite/btcd/chaincfg (from $GOROOT)
        /go/src/github.com/btcsuite/btcd/chaincfg (from $GOPATH)
breacharbiter.go:12:2: cannot find package "github.com/btcsuite/btcd/chaincfg/chainhash" in any of:
Step 5/12 : RUN apk add --no-cache     git     make &&  cd /go/src/github.com/lightningnetwork/lnd &&  make &&  make install
 ---> Running in 8d04c3b0e864
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
(1/7) Installing nghttp2-libs (1.32.0-r0)
(2/7) Installing libssh2 (1.8.0-r3)
(3/7) Installing libcurl (7.61.1-r1)
(4/7) Installing expat (2.2.5-r0)
(5/7) Installing pcre2 (10.31-r0)
(6/7) Installing git (2.18.1-r0)
(7/7) Installing make (4.2.1-r2)
Executing busybox-1.28.4-r1.trigger
OK: 19 MiB in 21 packages
\033[0;32m Building debug lnd and lncli.\033[0m
GO111MODULE=on go build -v -tags="dev" -o lnd-debug -ldflags "-X github.com/lightningnetwork/lnd/build.Commit=v0.5.1-beta-191-g0fafd5e2fd824f38ec6a03a56488de9c0798f34f" github.com/lightningnetwork/lnd
nat/upnp.go:9:2: cannot find package "github.com/NebulousLabs/go-upnp" in any of:
        /usr/local/go/src/github.com/NebulousLabs/go-upnp (from $GOROOT)
        /go/src/github.com/NebulousLabs/go-upnp (from $GOPATH)
aezeed/cipherseed.go:12:2: cannot find package "github.com/Yawning/aez" in any of:
        /usr/local/go/src/github.com/Yawning/aez (from $GOROOT)
        /go/src/github.com/Yawning/aez (from $GOPATH)
breacharbiter.go:11:2: cannot find package "github.com/btcsuite/btcd/blockchain" in any of:
        /usr/local/go/src/github.com/btcsuite/btcd/blockchain (from $GOROOT)
        /go/src/github.com/btcsuite/btcd/blockchain (from $GOPATH)
fundingmanager.go:11:2: cannot find package "github.com/btcsuite/btcd/btcec" in any of:
        /usr/local/go/src/github.com/btcsuite/btcd/btcec (from $GOROOT)
        /go/src/github.com/btcsuite/btcd/btcec (from $GOPATH)
chainntnfs/interface.go:7:2: cannot find package "github.com/btcsuite/btcd/btcjson" in any of:
        /usr/local/go/src/github.com/btcsuite/btcd/btcjson (from $GOROOT)
        /go/src/github.com/btcsuite/btcd/btcjson (from $GOPATH)
chainparams.go:4:2: cannot find package "github.com/btcsuite/btcd/chaincfg" in any of:
        /usr/local/go/src/github.com/btcsuite/btcd/chaincfg (from $GOROOT)
        /go/src/github.com/btcsuite/btcd/chaincfg (from $GOPATH)
breacharbiter.go:12:2: cannot find package "github.com/btcsuite/btcd/chaincfg/chainhash" in any of:
[...]
macaroons/constraints.go:10:2: cannot find package "gopkg.in/macaroon-bakery.v2/bakery/checkers" in any of:
        /usr/local/go/src/gopkg.in/macaroon-bakery.v2/bakery/checkers (from $GOROOT)
        /go/src/gopkg.in/macaroon-bakery.v2/bakery/checkers (from $GOPATH)
macaroons/auth.go:8:2: cannot find package "gopkg.in/macaroon.v2" in any of:
        /usr/local/go/src/gopkg.in/macaroon.v2 (from $GOROOT)
        /go/src/gopkg.in/macaroon.v2 (from $GOPATH)
make: *** [Makefile:100: build] Error 1
ERROR: Service 'lnd' failed to build: The command '/bin/sh -c apk add --no-cache     git     make &&  cd /go/src/github.com/lightningnetwork/lnd &&  make &&  make install' returned a non-zero code: 2
```

### CODE REVIEW

**Comment by yancyribbens** [comment]
I also submitted a PR for this issue: https://github.com/lightningnetwork/lnd/pull/2259/files

File: docker/README.md [hunk 1]
```diff
@@ -64,21 +64,21 @@ bitcoin into.
 $ export NETWORK="simnet" 
 
 # Run the "Alice" container and log into it:
-$ docker-compose run -d --name alice lnd_btc
-$ docker exec -i -t alice bash
+$ sudo -E docker-compose run -d --name alice lnd_btc
```
Comment (yancyribbens) [inline docker/README.md]: `sudo` isn't required although it may be required depending on your OS and security configuration.

Comment (yancyribbens) [reply docker/README.md]: same with the `-E` flag.  This is OS environment dependent.

Comment (tuxx42) [reply docker/README.md]: I just noticed there is also a `docker` command in that README not just `docker-compose`. The only time you dont need root when calling docker is if your user is in the `docker` group, which is a catastrophe from a security standpoint. It effectively gives your user passwordless sudo

Comment (yancyribbens) [reply docker/README.md]: @tuxx42 OSX does not require `sudo` and therefore doesn't require `-E`.  Granted this is most likely because the OSX implementation uses a guest kernel.  I think we could just make a note that `sudo -E` may be required without explicitly demanding the user elevate privilage.

File: docker/lnd/Dockerfile [hunk 2]
```diff
@@ -11,6 +11,7 @@ ENV GODEBUG netdns=cgo
 
 # Install dependencies and install/build lnd.
 RUN apk add --no-cache \
+    build-base \
```
Comment (yancyribbens) [inline docker/lnd/Dockerfile]: why `build-base` instead of `--update alpine-sdk` and `gcc`?  Not that I have a reason for one versus the other, i'm just curious what the reasoning is.

File: docker/btcd/start-btcd.sh [hunk 3]
```diff
@@ -44,8 +44,12 @@ RPCPASS=$(set_default "$RPCPASS" "devpass")
 DEBUG=$(set_default "$DEBUG" "info")
 NETWORK=$(set_default "$NETWORK" "simnet")
 
-PARAMS=$(echo \
-    "--$NETWORK" \
+PARAMS=""
+if [ "$NETWORK" != "mainnet" ]; then
```
Comment (yancyribbens) [inline docker/btcd/start-btcd.sh]: this seems like a separate issue and maybe should be a different PR?  `start-btcd` is also in need of updating https://github.com/lightningnetwork/lnd/blob/master/docker/btcd/start-btcd.sh#L48

Comment (yancyribbens) [reply docker/btcd/start-btcd.sh]: https://github.com/lightningnetwork/lnd/issues/2337

File: docker/README.md (hunk 1 above)
Comment (tuxx42) [reply docker/README.md]: agreed.

File: docker/lnd/Dockerfile (hunk 2 above)
Comment (tuxx42) [reply docker/lnd/Dockerfile]: no, probably your suggestion is better.

**Comment by valentinewallace** [comment]
hey @tuxx42! thanks for the submission. Your last commit looks like a very viable fix, would you mind amending it to include a fix for [btcctl](https://github.com/lightningnetwork/lnd/blob/master/docker/btcd/start-btcctl.sh) as well?

**Comment by Roasbeef** [comment]
Closing this as other PRs have superseded it.
